program/src/processor.rs

```diff
@@ -579,8 +609,35 @@ pub fn withdraw_excess_inactive_sol<'a, 'b>(
     Ok(excess_balance)
 }
 
-/// Recover any inactive balance from the validator's stake accounts to the
-/// reserve account.
+pub fn get_stake_account_balance(
+    stake_account_address: &Pubkey,
```

Comment:
Maybe we should make this a method on `Validator` instead of taking the address; right now you can just pass `accountinfo.key, accountinfo` as the arguments. (Even if it is not called like that right now, there is a risk that somebody will in the future.) If we instead take only the seed, and derive the expected address, then it’s much harder to use incorrectly.

Or maybe just call it `derived_stake_account_address` to at least clarify that we shouldn’t pass in just any address.

Reply:
Let's called it `derived_stake_account_address`, we can change it later